src/indexes/text/lexer.cc

```diff
@@ -8,48 +8,91 @@
 
 #include "absl/log/check.h"
 #include "absl/status/status.h"
+#include "absl/strings/ascii.h"
 #include "libstemmer.h"
+#include "src/utils/scanner.h"
 
 namespace valkey_search::indexes::text {
 
-// TODO : Update Constructor and Deconstructor below after rest of lexical operations are added
-Lexer::Lexer() : stemmer_(nullptr) {}
+absl::StatusOr<std::vector<std::string>> Lexer::Tokenize(
+    absl::string_view text,
+    const std::bitset<256>& punct_bitmap,
+    sb_stemmer* stemmer,
+    bool stemming_enabled,
+    uint32_t min_stem_size) const {
 
-Lexer::~Lexer() {
-  if (stemmer_) {
-    sb_stemmer_delete(stemmer_);
+  if (!IsValidUtf8(text)) {
+    return absl::InvalidArgumentError("Invalid UTF-8");
   }
-}
 
-absl::Status Lexer::Initialize(const std::string& language) {
-  if (stemmer_) {
-    sb_stemmer_delete(stemmer_);
+  std::vector<std::string> tokens;
+
+  size_t pos = 0;
+  while (pos < text.size()) {
+    while (pos < text.size() &&
+           punct_bitmap[static_cast<unsigned char>(text[pos])]) {
+      pos++;
+    }
+
+    size_t word_start = pos;
+    while (pos < text.size() &&
+           !punct_bitmap[static_cast<unsigned char>(text[pos])]) {
+      pos++;
+    }
+
+    if (pos > word_start) {
+      std::string word(text.data() + word_start, pos - word_start);
+
+      word = absl::AsciiStrToLower(word);
+
+      // TODO: Stop word removal
+
+      if (stemming_enabled && word.length() >= min_stem_size) {
```

Comment:
Can we handle this logic (`stemming_enabled && word.length() >= min_stem_size`) inside the `StemWord` function? It will help the caller sites such as the query parsing logic